npm/qsharp/ux/scatterChart.tsx

```diff
@@ -154,6 +156,12 @@ export function ScatterChart(props: {
   }
   const selectedPoint = getSelectedPointData();
 
+  const saveRef = createRef();
+
+  const handleSaveImage = async () => {
+    saveRefToImage(saveRef);
+  };
```

Comment:
You'll probably want to use `useRef` here instead (as we do in other places) .

```suggestion
  const saveRef = useRef<SVGSVGElement>(null);

  const handleSaveImage = async () => {
    saveToImage(saveRef.current!);
  };
```

Also would recommend calling  `.current` here at the call site, so that `saveToImage()` takes the element instead of the ref. Makes the code boundaries cleaner, IMO. (I renamed `saveRefToImage() -> saveToImage()` to make it clear it doesn't take a ref)

Reply:
I tried `useRef` instead of `createRef` earlier and ran into an error. So I looked around and found [this post](https://stackoverflow.com/a/61638527) which shows how they are related. This is why I chose to use `createRef` which would also avoid type casting which we would need to do otherwise. Quick and ugly way to do this (at the call site or somewhere else:
```
    saveToImage(saveRef.current! as unknown as HTMLElement);
```
We cannot cast directly from `SVGSVGElement` as it does not inherit from `HTMLElement`. However, this code smells so I found an overly complicated route that uses a `DOMParser` to convert to string first, then appends to a newly created `HTMLElement`.

Overall I suggest we continue to use `createRef`, it does not have these issues.

Reply:
Couldn't `saveToImage` just take a `SVGSVGElement` directly? There's really not a big need to make it that generic. I'm still a bit iffy about using `createRef` (personally having a strong bias towards consistency).

I agree about wanting to avoid the cast. If there's no other way to make it work it's ok.

Reply:
I don't know of another way, tried casting `SVGSVGElement` to `Element` but it lacks a `style` prop which belongs to `HTMLElement`. This prop is required when we clone the style.